src/mesh/analysis.rs

```diff
@@ -242,14 +303,18 @@ pub fn border_edge_loops(edge_sharing: &EdgeSharingMap) -> Vec<Vec<OrientedEdge>
             }
         };
 
-        // There can theoretically exist chains of edges that do not form border
-        // edge loops, e.g. if the mesh is non-manifold. We do not output those.
-        if loop_closed {
-            edge_loops.push(Vec::from(current_chain));
+        edge_loops.push(Vec::from(current_chain));
+
+        if !loop_closed {
+            non_deterministic = true;
```

Comment:
Not sure I agree with this change. If the loop wasn't closed, it is not a loop. Should we be including it? And if we do want to include it, isn't this another edge-case or another case of non-determinism that is distinct from the one we detect by counting vertex index occurrences? Shouldn't we tell that, for example with another result variant that returns a vector of closed loops and a vector of unclosed loops? Something like:

```rust
pub enum BorderEdgeLoopsResult {
    Found {
        detected_nondeterminism: bool,
        loops: Vec<Vec<OrientedEdge>>,
    }
    FoundWithIncompleteLoops {
        detected_nondeterminism: bool,
        loops_closed: Vec<Vec<OrientedEdge>>,
        loops_unclosed: Vev<Vec<OrientedEdge>>,
    }
    Watertight,
}
```

Reply:
is it even possible to get non-closed loops? what was the case with two-segment long edge loops? can we replicate it?

Reply:
I didn't really find out. The geometry was huge and the original bug could have theoretically come from elsewhere. We could checkout the original infinite looping function and find out, if we really want to. I didn't try reproducing on smaller data, but in theory, we could hand-craft such data. Do you want to do it or should I? Either way, we should know what the assumptions are. If such data is impossible, we should be able to prove that. If it is possible, we need to handle the posibility.

Reply:
so, what do you prefer? i'd be also ok with panicking if the loop is not closed. it means the input data wasn't valid.

Reply:
If we truly believe this cannot happen to our mesh, then let's assert.

However, we can only panic if we are asserting certain invariants hold. In this case, the invariant that a border edge loop must be closed. This is true only if the data comes from a real mesh, but currently it doesn't have to, because the API allows you to hand-craft edge sharing maps.

We need to change the API and before we can safely assert (not necessarily in this order - see https://github.com/subdgtl/HURBAN-Selector/pull/171/files#r404086223)

Reply:
This is a more general problem - we are almost nowhere sure the data comes from any valid source. Shouldn't we make a new task to change the API so that this is safe?

Reply:
let's assert this shouldn't happen. if it does, we messed up somewhere because we let it happen.